source/gear/swordofthesun-thiefs.js

```diff
@@ -0,0 +1,48 @@
+const { GearTemplate } = require('../classes');
+const { ESSENCE_MATCH_STAGGER_FOE } = require('../constants');
+const { getModifierCategory } = require('../modifiers/_modifierDictionary');
+const { dealDamage, changeStagger, combineModifierReceipts, removeModifier, generateModifierResultLines } = require('../util/combatantUtil');
+const { damageScalingGenerator } = require('./shared/scalings');
+
+module.exports = new GearTemplate("Thief's Sword of the Sun",
+	[
+		["use", "Removes all buffs from a foe, then deal <@{damage}> @{essence} damage to a them, plus 30 damage per unique buff removed, then gain @{bounty}g if they're downed. This is a habit, isn't it?"],
+		["critical", "Damage x @{critBonus}"]
+	],
+	"Offense",
+	"Fire"
+).setCost(450)
+	.setEffect((targets, user, adventure) => {
+		totalResultLines = [];
+		for (const target of targets) {
+			const targetBuffs = Object.keys(target.modifiers).filter(modifier => getModifierCategory(modifier) === "Buff");
+			removedBuffReceipts = []
+			targetBuffs.forEach(buffName => {
+				removedBuffReceipts.push(...removeModifier([target], { name: buffName, stacks: "all" }))
+			})
+			totalResultLines.push(...generateModifierResultLines(combineModifierReceipts(removedBuffReceipts)));
+
+			const { essence, scalings: { damage, critBonus, bounty } } = module.exports;
+			let pendingDamage = damage.calculate(user) + 30 * targetBuffs.length;
+			if (user.crit) {
+				pendingDamage *= critBonus;
+			}
+			const { resultLines, survivors } = dealDamage([target], user, pendingDamage, false, essence, adventure);
+			totalResultLines.push(...resultLines)
+			if (user.essence === essence) {
+				changeStagger(survivors, user, ESSENCE_MATCH_STAGGER_FOE);
+			}
+			if (survivors.length < 1) {
+				adventure.room.addResource("Gold", "Currency", "loot", bounty);
+				totalResultLines.push(`${user.name} pillages ${bounty}g.`);
+			}
+		}
+		return totalResultLines;
+	}, { type: "single", team: "foe" })
+	.setSidegrades("Lethal Sword of the Sun")
+	.setCooldown(3)
+	.setScalings({
+		damage: damageScalingGenerator(40),
```

Comment:
I think it's worth to make a custom `swordOfTheSunDamageScalingGenerator()` so that we can change the scaling on all upgrades in one place.

Reply:
err.... actually I don't think I understand what exactly a damageScalingGenerator is or how it'd factor out

Reply:
In `source/gear/shared/scalings.js` (same place `damageScalingGenerator()` is imported from), we have some functions that are customized versions of `damageScalingGenerator()`. A customized version for Sword of the Sun would allow us to tweak that scaling on all SotS variants in that one file without risk of forgetting to upgrade one of the variants.

Reply:
Realizing I just repeated myself and didn't really answer the question: a `Scaling` is a relatively recently introduced class that contains information about how damage or another value changes with the user's stats. It has both a string description for use in description generation and a calculate function that outputs the number described by the description when given the user Combatant.

While adapting those to use in Gear, I found that I'd really need families of `Scaling`s instead of specific instances, so the generators are functions that make `Scaling`s based on whatever input they're provided.